Index: tisyslink-ducati-0.0+cc+BRIDGE-DUCATI_RLS_0.13/ducati_cfg/DucatiAmmu.cfg
===================================================================
--- tisyslink-ducati-0.0+cc+BRIDGE-DUCATI_RLS_0.13.orig/ducati_cfg/DucatiAmmu.cfg	2009-05-15 02:34:42.000000000 -0500
+++ tisyslink-ducati-0.0+cc+BRIDGE-DUCATI_RLS_0.13/ducati_cfg/DucatiAmmu.cfg	2009-12-02 13:52:00.000000000 -0600
@@ -38,7 +38,10 @@
  */
  
 Cache = xdc.useModule('ti.sysbios.hal.unicache.Cache');
-Cache.enableCache = true;
+Cache.enableCache = false
+
+var hCache = xdc.useModule('ti.sysbios.hal.Cache');
+hCache.CacheProxy = xdc.useModule('ti.sysbios.hal.CacheNull');
 
 var AMMU = xdc.useModule('ti.sysbios.hal.ammu.AMMU');
 
