--- meta/packages/e2fsprogs/e2fsprogs_1.41.2.bb
+++ meta/packages/e2fsprogs/e2fsprogs_1.41.2.bb
@@ -10,6 +10,12 @@ EXTRA_OECONF += " --sbindir=${base_sbindir} --enable-elf-shlibs"
 EXTRA_OECONF_darwin = "--enable-dynamic-e2fsck --sbindir=${base_sbindir} --enable-bsd-shlibs"
 EXTRA_OECONF_darwin8 = "--enable-dynamic-e2fsck --sbindir=${base_sbindir} --enable-bsd-shlibs"
 
+do_configure_prepend() {
+	autopoint -f
+	echo "install-shlibs:" >> intl/Makefile.in
+	echo "" >> intl/Makefile.in
+}
+
 do_compile_prepend () {
    find ./ -print|xargs chmod u=rwX
    ( cd util; ${BUILD_CC} subst.c -o subst )
--
1.5.6.3

