--- meta/packages/netbase/netbase_4.21.bb
+++ meta/packages/netbase/netbase_4.21.bb
@@ -38,7 +38,7 @@ do_install () {
 
 	# Disable network manager on machines that commonly do NFS booting
 	case "${MACHINE}" in
-		"omap-3430sdp" | "omap-3430ldp" | "omap-2430sdp" | "qemuarm" | "qemux86" )
+		"omap-4430sdp" | "omap-4430sdphigh" | "omap-3430sdp" | "omap-3430ldp" | "omap-2430sdp" | "qemuarm" | "qemux86" )
 			touch ${D}${sysconfdir}/network/nm-disabled-eth0
 			;;
 		*)
