Index: git/CUDK/os/linux/src/makefile
===================================================================
--- git.orig/CUDK/os/linux/src/makefile	2009-12-22 11:02:48.000056000 +0100
+++ git/CUDK/os/linux/src/makefile	2009-12-22 11:03:09.000019000 +0100
@@ -122,7 +122,7 @@
 
 $(OUTPUT_DIR)/$(LIB_NAME).a: $(OBJS)
 	rm -f $@
-	$(CROSS_COMPILE)$(AR) -q $@ $^
+	$(AR) -q $@ $^
 
 $(OUTPUT_DIR)/$(LIB_NAME).so: $(OBJS)
 	rm -f $@
Index: git/platforms/os/common/build/linux/Makefile
===================================================================
--- git.orig/platforms/os/common/build/linux/Makefile	2009-12-22 11:03:24.000064000 +0100
+++ git/platforms/os/common/build/linux/Makefile	2009-12-22 11:04:09.000019000 +0100
@@ -129,7 +129,7 @@
 #
 $(OUTPUT_FILE):
 	$(MAKE) -C $(KERNEL_DIR) M=`pwd` ARCH=arm CROSS_COMPILE=$(CROSS_COMPILE) modules
-	@$(CROSS_COMPILE)$(AR) rcs $@ $(OS_OBJS)
+	@$(AR) rcs $@ $(OS_OBJS)
 
 
 endif	# ifneq ($(KERNELRELEASE),)
Index: git/stad/build/linux/Makefile
===================================================================
--- git.orig/stad/build/linux/Makefile	2009-12-22 11:04:41.000046000 +0100
+++ git/stad/build/linux/Makefile	2009-12-22 11:04:54.000019000 +0100
@@ -125,7 +125,7 @@
 #
 $(OUTPUT_FILE):
 	$(MAKE) -C $(KERNEL_DIR) M=`pwd` ARCH=arm CROSS_COMPILE=$(CROSS_COMPILE) modules
-	@$(CROSS_COMPILE)$(AR) rcs $@ $(DK_OBJS)
+	@$(AR) rcs $@ $(DK_OBJS)
 
 
 endif	# ifneq ($(KERNELRELEASE),)
