Index: git/CUDK/configurationutility/makefile
===================================================================
--- git.orig/CUDK/configurationutility/makefile	2010-02-16 09:07:43.000038000 +0100
+++ git/CUDK/configurationutility/makefile	2010-02-16 09:08:09.000005000 +0100
@@ -99,6 +99,7 @@
 	-I $(WILINK_ROOT)/platforms/os/linux/inc \
 	-I $(WILINK_ROOT)/platforms/os/common/inc \
 	-I $(KERNEL_DIR)/include \
+	-I $(KERNEL_DIR)/arch/arm/include \
 	-I $(WILINK_ROOT)/TWD/FW_Transfer/Export_Inc \
     -I $(TI_SUPP_LIB_DIR)  
 
