Index: gst-plugins-good-0.10.7/gst/multifile/gstmultifilesink.c
===================================================================
--- gst-plugins-good-0.10.7.orig/gst/multifile/gstmultifilesink.c	2008-02-07 21:44:12.000000000 -0600
+++ gst-plugins-good-0.10.7/gst/multifile/gstmultifilesink.c	2009-11-04 18:43:26.000000000 -0600
@@ -202,6 +202,40 @@
   return filename;
 }
 
+void
+gst_multi_file_sink_post_timestamp(GstBaseSink* self,
+							gchar* structure_name)
+{
+
+  GstMessage *msg = NULL;
+  GstStructure *stru = NULL;
+  gdouble timestamp;
+  gchar *str_timestamp;
+  GTimeVal current_time;
+
+  /* Get the current time */
+  g_get_current_time(&current_time);
+  timestamp = current_time.tv_sec + current_time.tv_usec/1e6;
+
+  str_timestamp = g_strdup_printf("%f",timestamp);
+
+  if(structure_name)
+  {
+	stru = gst_structure_new(structure_name,
+						     "timestamp",
+						     G_TYPE_STRING,
+						     str_timestamp,
+						     NULL);
+  }
+
+  /* Create the message and send it to the bus */
+  msg = gst_message_new_custom (GST_MESSAGE_ELEMENT, GST_OBJECT (self), stru);
+
+  g_assert(gst_element_post_message (GST_ELEMENT(self), msg));
+  GST_INFO_OBJECT(self, "Sended messge with a timestamp %f", timestamp);
+  return;
+}
+
 static GstFlowReturn
 gst_multi_file_sink_render (GstBaseSink * sink, GstBuffer * buffer)
 {
@@ -222,6 +256,8 @@
   if (ret) {
     multifilesink->index++;
     g_free (filename);
+    gst_multi_file_sink_post_timestamp (GST_ELEMENT (sink),
+							"save-image");
     return GST_FLOW_OK;
   }
 
