diff --git a/sys/v4l2/gstv4l2object.c b/sys/v4l2/gstv4l2object.c
index 144e025..3512ada 100644
--- a/sys/v4l2/gstv4l2object.c
+++ b/sys/v4l2/gstv4l2object.c
@@ -1097,8 +1097,10 @@ gst_v4l2_object_v4l2fourcc_to_structure (guint32 fourcc)
           g_assert_not_reached ();
           break;
       }
-      structure = gst_structure_new ("video/x-raw-yuv",
-          "format", GST_TYPE_FOURCC, fcc, NULL);
+      structure = gst_structure_new ("video/x-raw-yuv-strided",
+          "format", GST_TYPE_FOURCC, fcc,
+          "rowstride", G_TYPE_INT, 4096,
+          NULL);
       break;
     }
     case V4L2_PIX_FMT_DV:
@@ -1219,7 +1221,7 @@ gst_v4l2_object_get_caps_info (GstV4l2Object * v4l2object, GstCaps * caps,
   *fps_n = gst_value_get_fraction_numerator (framerate);
   *fps_d = gst_value_get_fraction_denominator (framerate);
 
-  if (!strcmp (mimetype, "video/x-raw-yuv")) {
+  if (!strcmp (mimetype, "video/x-raw-yuv-strided")) {
     gst_structure_get_fourcc (structure, "format", &fourcc);
 
     switch (fourcc) {
