From 2133ca3318acea1065d9e88db49b8d8b3740f42e Mon Sep 17 00:00:00 2001
From: Rob Clark <rob@ti.com>
Date: Mon, 16 Nov 2009 21:26:17 -0600
Subject: [PATCH 08/12] omap4:  for now use OMX_AllocateBuffer() by default
To: gst_omap4@list.ti.com

Signed-off-by: Rob Clark <rob@ti.com>
---
 omx/gstomx_base_filter.c |    6 ++++--
 1 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/omx/gstomx_base_filter.c b/omx/gstomx_base_filter.c
index 5b79791..2ee7596 100644
--- a/omx/gstomx_base_filter.c
+++ b/omx/gstomx_base_filter.c
@@ -77,8 +77,10 @@ setup_ports (GstOmxBaseFilter *self)
     }
     else
     {
-        self->in_port->share_buffer = TRUE;
-        self->out_port->share_buffer = TRUE;
+        self->in_port->omx_allocate = TRUE;
+        self->out_port->omx_allocate = TRUE;
+        self->in_port->share_buffer = FALSE;
+        self->out_port->share_buffer = FALSE;
     }
 
     GST_DEBUG_OBJECT (self, "in_port->omx_allocate=%d, out_port->omx_allocate=%d",
-- 
1.6.3.2

