From 2937f70b81ec999497873423efd2694e769cb38d Mon Sep 17 00:00:00 2001
From: Rob Clark <rob@ti.com>
Date: Thu, 19 Nov 2009 15:23:30 -0600
Subject: [PATCH 9/9] remove omap3 specific custom index's
To: gst_omap4@list.ti.com

---
 omx/gstomx_jpegdec.c |    5 ++++-
 1 files changed, 4 insertions(+), 1 deletions(-)

diff --git a/omx/gstomx_jpegdec.c b/omx/gstomx_jpegdec.c
index ce2b41f..decb92c 100644
--- a/omx/gstomx_jpegdec.c
+++ b/omx/gstomx_jpegdec.c
@@ -244,7 +244,6 @@ omx_setup (GstOmxBaseFilter *omx_base)
     GOmxCore *gomx;
     gint width, height;
     OMX_COLOR_FORMATTYPE color_format;
-    OMX_INDEXTYPE index;
 
     self = GST_OMX_JPEGDEC (omx_base);
     gomx = (GOmxCore *) omx_base->gomx;
@@ -303,8 +302,10 @@ omx_setup (GstOmxBaseFilter *omx_base)
     }
 
     /*Set parameters*/
+#ifdef OMAP3
     {
         OMX_CUSTOM_RESOLUTION pMaxResolution;
+        OMX_INDEXTYPE index;
 #if 0
         /*By the moment properties don't have been added */
         OMX_CUSTOM_IMAGE_DECODE_SECTION pSectionDecode;
@@ -362,6 +363,7 @@ omx_setup (GstOmxBaseFilter *omx_base)
 
     /*Set config*/
     {
+        OMX_INDEXTYPE index;
         OMX_U32 nProgressive;
         /*Dinamic color change */
         OMX_GetExtensionIndex(gomx->omx_handle, "OMX.TI.JPEG.decode.Config.OutputColorFormat", &index);
@@ -376,6 +378,7 @@ omx_setup (GstOmxBaseFilter *omx_base)
         OMX_SetConfig(gomx->omx_handle, index, &(nProgressive));
 
     }
+#endif
     GST_INFO_OBJECT (omx_base, "end");
 }
 
-- 
1.6.3.2

