Index: git/scripts/setlocalversion
===================================================================
--- git.orig/scripts/setlocalversion	2010-04-09 09:46:32.000000000 +0200
+++ git/scripts/setlocalversion	2010-04-09 09:47:08.000000000 +0200
@@ -17,7 +17,9 @@
 cd "${1:-.}" || usage
 
 # Check for git and a git repo.
-if head=`git rev-parse --verify --short HEAD 2>/dev/null`; then
+# Hack: into poky, kernel is not a git tree
+#if head=`git rev-parse --verify --short HEAD 2>/dev/null`; then
+if [ "0" = "1" ]; then
 
 	# If we are at a tagged commit (like "v2.6.30-rc6"), we ignore it,
 	# because this version is defined in the top level Makefile.
