--- a/src/eurasia/eurasiacon/build/linux/makefile.conf
+++ b/src/eurasia/eurasiacon/build/linux/makefile.conf
@@ -80,7 +80,7 @@
 
 # NEED TO CHECK VERSION OF BISON
 #
-REQUIREDBISONVERSION = 2.3
+REQUIREDBISONVERSION = 2.0
 BISONVERSION		 = $(shell $(BISON) --version | $(GREP) "bison (GNU Bison)" | cut -f4 -d' ')
 a;lsdsa;ldasl;kdlsa;kdal;kdl;ka;lkasdlk
 # This macro represents a file in which is placed the date, the name of the

