Index: git/eurasia/eurasiacon/build/linux/omap3630_linux/makefile.conf
===================================================================
--- git.orig/eurasia/eurasiacon/build/linux/omap3630_linux/makefile.conf	2010-04-09 12:22:20.000000000 +0200
+++ git/eurasia/eurasiacon/build/linux/omap3630_linux/makefile.conf	2010-04-09 12:22:27.000000000 +0200
@@ -28,7 +28,7 @@
 
 X11ROOT			=	/usr/X11R6_SGX
 
-SUPPORT_XWS  = 1
+SUPPORT_XWS  = 0
 SUPPORT_EWS  = 1
 PROFILE_COMMON = 1
 
