Index: wpa_supplicant-0.5.11/driver_wext.c
===================================================================
--- wpa_supplicant-0.5.11.orig/driver_wext.c	2010-01-26 22:14:16.001463000 +0100
+++ wpa_supplicant-0.5.11/driver_wext.c	2010-01-26 22:18:52.000159000 +0100
@@ -1001,6 +1001,11 @@
 {
 	int flags;
 
+        if (wpa_driver_wext_get_ifflags(drv, &flags) != 0 || wpa_driver_wext_set_ifflags(drv, flags | IFF_UP) != 0) { 
+                printf("Could not set interface '%s' UP\n", drv->ifname);
+        }
+        os_sleep(0, 400000);
+
 	/*
 	 * Make sure that the driver does not have any obsolete PMKID entries.
 	 */
