Index: git/configure.ac
===================================================================
--- git.orig/configure.ac	2010-04-22 18:50:18.000000000 -0500
+++ git/configure.ac	2010-04-23 13:16:11.000000000 -0500
@@ -313,14 +313,14 @@
 AG_GST_SET_ERROR_CFLAGS($GST_GIT, [
     -Wmissing-declarations -Wmissing-prototypes -Wredundant-decls -Wundef
     -Wwrite-strings -Wformat-nonliteral -Wformat-security
-    -Winit-self -Wmissing-include-dirs -Waddress -Waggregate-return
+    -Winit-self -Waddress -Waggregate-return
     -Wno-multichar -Wnested-externs ])
 
 dnl define an ERROR_CXXFLAGS Makefile variable
 AG_GST_SET_ERROR_CXXFLAGS($GST_GIT, [
     -Wmissing-declarations -Wredundant-decls -Wundef
     -Wwrite-strings -Wformat-nonliteral -Wformat-security
-    -Winit-self -Wmissing-include-dirs -Waddress -Waggregate-return
+    -Winit-self -Waddress -Waggregate-return
     -Wno-multichar])
 
 dnl define correct level for debugging messages
