--- a/WTSD_DucatiMMSW/framework/domx_linux_sample/src/make/build.mk
+++ b/WTSD_DucatiMMSW/framework/domx_linux_sample/src/make/build.mk
@@ -66,7 +66,7 @@
 CFLAGS += -fomit-frame-pointer
 endif
 #   Warnings
-CFLAGS += -Wall  -Wno-trigraphs -Werror-implicit-function-declaration #-Wno-format
+CFLAGS += -Wall  -Wno-trigraphs #-Werror-implicit-function-declaration -Wno-format
 #   Optimizations
 #CFLAGS += -O2 -fno-strict-aliasing
 #CFLAGS += -Os -fno-strict-aliasing
