--- a/WTSD_DucatiMMSW/framework/domx_linux_sample/src/system/domx/omx_proxy_common/src/omx_proxy_common.c
+++ b/WTSD_DucatiMMSW/framework/domx_linux_sample/src/system/domx/omx_proxy_common/src/omx_proxy_common.c
@@ -40,13 +40,13 @@
 #ifdef TILER_BUFF
 #include <ProcMgr.h>
 #include <SysLinkMemUtils.h>
-#include "memmgr/mem_types.h"
+#include "tiler/mem_types.h"
 //#include "utils/utils.h"
 //#include "debug_utils.h"
 //#include "list_utils.h"
 //#include "tilermem_utils.h"
-#include "d2c/phase1_d2c_remap.h"
-#include "memmgr/memmgr.h"
+#include "d2cmap/phase1_d2c_remap.h"
+#include "tiler/memmgr.h"
 //#include "tiler.h"
 #define PortFormatIsNotYUV 0
 #endif--- a/WTSD_DucatiMMSW/framework/domx_linux_sample/test/h264dec/src/H264_Decoder_ILClient.c
+++ b/WTSD_DucatiMMSW/framework/domx_linux_sample/test/h264dec/src/H264_Decoder_ILClient.c
@@ -74,7 +74,7 @@
    #endif
    
    #ifdef OMX_H264D_LINUX_TILERTEST
-      #include <memmgr/memmgr.h>
+      #include <tiler/memmgr.h>
       #include <mmplatform.h>
    #endif
    #ifdef OMX_H264D_BUF_HEAP--- a/WTSD_DucatiMMSW/framework/domx_linux_sample/test/sample_proxy/test_sample_proxy.c
+++ b/WTSD_DucatiMMSW/framework/domx_linux_sample/test/sample_proxy/test_sample_proxy.c
@@ -56,8 +56,8 @@
 
 #ifdef OMX_SAMPLE_TILER_TEST 
 
-#include <memmgr/memmgr.h>
-#include <memmgr/mem_types.h>
+#include <tiler/memmgr.h>
+#include <tiler/mem_types.h>
 
  #define OMX_SAMPLE_IN_2DYHEIGHT 4
  #define OMX_SAMPLE_IN_2DYWIDTH 8--- a/WTSD_DucatiMMSW/framework/domx_linux_sample/test/h264enc/src/H264_ILClient.c
+++ b/WTSD_DucatiMMSW/framework/domx_linux_sample/test/h264enc/src/H264_ILClient.c
@@ -45,7 +45,7 @@
 
 #ifdef OMX_H264E_LINUX_TILERTEST
 /*Tiler APIs*/
-#include <memmgr/memmgr.h>
+#include <tiler/memmgr.h>
 #include <mmplatform.h>
 
 #endif--- a/WTSD_DucatiMMSW/framework/domx_linux_sample/test/omtb/src/omtbh264vdec.c
+++ b/WTSD_DucatiMMSW/framework/domx_linux_sample/test/omtb/src/omtbh264vdec.c
@@ -3,7 +3,7 @@
 \******************************************************************************/
 /** @file      omtbH264Vdec.c
  *
- *  @brief     H264 Video Decode IL Client Implementation.
+ *  @brief     H264 Video Decode IL Client Implementation.me
  *
  * <B> History: </B>
  *
@@ -49,7 +49,7 @@
 #endif
 
 #ifdef TILER
-#include <memmgr/memmgr.h>
+#include <tiler/memmgr.h>
 #include <mmplatform.h>
 #endif
 /******************************************************************************\
--- a/WTSD_DucatiMMSW/framework/domx_linux_sample/test/omtb/src/omtbmain.c
+++ b/WTSD_DucatiMMSW/framework/domx_linux_sample/test/omtb/src/omtbmain.c
@@ -53,7 +53,7 @@
 #include "omtbCmd.h"
 
 #ifdef WTSD
-#include <memmgr/memmgr.h>
+#include <tiler/memmgr.h>
 #include <mmplatform.h>
 #else
 /* Codec engine header files */
--- a/WTSD_DucatiMMSW/framework/domx_linux_sample/test/omtb/src/omtbh264venc.c
+++ b/WTSD_DucatiMMSW/framework/domx_linux_sample/test/omtb/src/omtbh264venc.c
@@ -49,7 +49,7 @@
 
 #ifdef TILER //OMX_H264E_LINUX_TILERTEST //Brijesh
 /*Tiler APIs*/
-#include <memmgr/memmgr.h>
+#include <tiler/memmgr.h>
 #include <mmplatform.h>
 #endif
 
--- a/WTSD_DucatiMMSW/framework/domx_linux_sample/test/jpegdec/src/omx_jpegd.c
+++ b/WTSD_DucatiMMSW/framework/domx_linux_sample/test/jpegdec/src/omx_jpegd.c
@@ -62,8 +62,8 @@
 
 #ifdef OMX_LINUX_TILERTEST
 /*Tiler APIs*/
-#include <memmgr/memmgr.h>
-#include <memmgr/tiler.h>
+#include <tiler/memmgr.h>
+#include <tiler/tiler.h>
 #endif
 
 FILE *op_file,*ref_op_file;--- a/WTSD_DucatiMMSW/framework/domx_linux_sample/test/jpegdec/src/omx_jpegd_test_main.c
+++ b/WTSD_DucatiMMSW/framework/domx_linux_sample/test/jpegdec/src/omx_jpegd_test_main.c
@@ -45,7 +45,7 @@
 
 #ifdef OMX_LINUX_TILERTEST
 /*Tiler APIs*/
-#include <memmgr/memmgr.h>
+#include <tiler/memmgr.h>
 #include <mmplatform.h>
 
 #endif--- a/WTSD_DucatiMMSW/framework/domx_linux_sample/test/camera/src/camera_il_client.c
--- b/WTSD_DucatiMMSW/framework/domx_linux_sample/test/camera/src/camera_il_client.c
@@ -88,7 +88,7 @@
 
 /*Tiler APIs*/
 #ifdef TILER_BUFFERS
-#include <memmgr/memmgr.h>
+#include <tiler/memmgr.h>
 #include <mmplatform.h>
 #define STRIDE_8BIT (4 * 1024)
 #define STRIDE_16BIT (4 * 1024)
