Index: libowl-av/libowl-av/owl-video-widget.c
===================================================================
--- libowl-av.orig/libowl-av/owl-video-widget.c	2008-08-23 09:16:54.000000000 -0500
+++ libowl-av/libowl-av/owl-video-widget.c	2009-11-29 15:30:27.000000000 -0600
@@ -330,7 +330,7 @@
                                     &format,
                                     &duration);
 
-        if (format != GST_FORMAT_TIME)
+        if ((format != GST_FORMAT_TIME) || (duration == -1L))
                 return;
 
         video_widget->priv->duration = duration / GST_SECOND;
@@ -433,7 +433,7 @@
          * playbin.
          **/
         video_widget->priv->playbin =
-                gst_element_factory_make ("playbin", "playbin");
+                gst_element_factory_make ("playbin2", "playbin2");
         if (!video_widget->priv->playbin) {
                 g_warning ("No playbin found. Playback will not work.");
 
Index: libowl-av/libowl-av/owl-audio-player.c
===================================================================
--- libowl-av.orig/libowl-av/owl-audio-player.c	2009-11-29 15:32:41.000000000 -0600
+++ libowl-av/libowl-av/owl-audio-player.c	2009-11-29 15:35:11.000000000 -0600
@@ -170,7 +170,7 @@
                                     &format,
                                     &duration);
 
-        if (format != GST_FORMAT_TIME)
+        if ((format != GST_FORMAT_TIME) || (duration == -1L))
                 return;
 
         audio_player->priv->duration = duration / GST_SECOND;
@@ -273,7 +273,7 @@
          * playbin.
          **/
         audio_player->priv->playbin =
-                gst_element_factory_make ("playbin", "playbin");
+                gst_element_factory_make ("playbin2", "playbin2");
         if (!audio_player->priv->playbin) {
                 g_warning ("No playbin found. Playback will not work.");
 
