Index: tiopenmax-3.20/application/src/openmax_il/avplay/src/Makefile
===================================================================
--- tiopenmax-3.20.orig/application/src/openmax_il/avplay/src/Makefile	2009-03-13 12:00:48.000000000 -0500
+++ tiopenmax-3.20/application/src/openmax_il/avplay/src/Makefile	2009-03-28 18:57:13.000000000 -0500
@@ -6,8 +6,7 @@
 
 SRC=\
 	AvPlayTest.c \
-	AvPlayTest_WMV.c \
-    AvPlayTest_Real.c
+	AvPlayTest_WMV.c
 
 HSRC=$(wildcard ../inc/*)
 
@@ -50,7 +49,7 @@
 	cp -f $(COMPONENT_TEST) $(COMPONENT_TARGET)
 
 $(COMPONENT_TEST): $(OBJ)
-	$(CROSS)gcc $(CFLAGS) -o $(COMPONENT_TEST) $(OBJ) -L$(OMXLIBDIR) -ldl -lOMX_Core -lpthread -lRM_Parser
+	$(CROSS)gcc $(CFLAGS) -o $(COMPONENT_TEST) $(OBJ) -L$(OMXLIBDIR) -ldl -lOMX_Core -lpthread 
 	
 $(SRC): $(HSRC)
 
