--- a/src/eurasia_km/services4/3rdparty/dc_omap3430_linux/omaplfb_displayclass.c
+++ b/src/eurasia_km/services4/3rdparty/dc_omap3430_linux/omaplfb_displayclass.c
@@ -29,7 +29,7 @@
 #include <linux/console.h>
 #include <linux/fb.h>
 
-#include <mach/vrfb.h>
+#include <plat/vrfb.h>
 #include <../drivers/video/omap2/omapfb/omapfb.h>
 
 #include <linux/module.h>
--- a/src/eurasia_km/services4/3rdparty/dc_omap3430_linux/omaplfb_linux.c
+++ b/src/eurasia_km/services4/3rdparty/dc_omap3430_linux/omaplfb_linux.c
@@ -53,7 +53,7 @@
 #include <asm/io.h>
 
 #if (LINUX_VERSION_CODE > KERNEL_VERSION(2,6,26))
-#include <mach/display.h>
+#include <plat/display.h>
 #else 
 #include <asm/arch-omap/display.h>
 #endif 
