--- a/WTSD_DucatiMMSW/alg/lbce/test/src/lbce_test_main.c
+++ b/WTSD_DucatiMMSW/alg/lbce/test/src/lbce_test_main.c
@@ -29,18 +29,18 @@
 #include<stdlib.h>
 #include<string.h>
 #include <ti/sysbios/BIOS.h>
-#include <wtsd_ducatimmsw/alg/lbce/test/inc/lbce_test_main.h>
-#include <wtsd_ducatimmsw/alg/ip_run/inc/ip_run.h>
-#include <wtsd_ducatimmsw/drivers/csl/iss/simcop/common/simcop.h>
-#include  <wtsd_ducatimmsw/drivers/csl/iss/iss_common/iss_common.h>
-#include <wtsd_ducatimmsw/drivers/csl/iss/simcop/common/simcop_irq.h>
-#include <wtsd_ducatimmsw/platform/osal/timm_osal_semaphores.h>
+#include <WTSD_DucatiMMSW/alg/lbce/test/inc/lbce_test_main.h>
+#include <WTSD_DucatiMMSW/alg/ip_run/inc/ip_run.h>
+#include <WTSD_DucatiMMSW/drivers/csl/iss/simcop/common/simcop.h>
+#include  <WTSD_DucatiMMSW/drivers/csl/iss/iss_common/iss_common.h>
+#include <WTSD_DucatiMMSW/drivers/csl/iss/simcop/common/simcop_irq.h>
+#include <WTSD_DucatiMMSW/platform/osal/timm_osal_semaphores.h>
 
-#include "wtsd_ducatimmsw/framework/msp/msp.h"
-#include "wtsd_ducatimmsw/framework/resource_manager/rm.h"
-#include  "wtsd_ducatimmsw/drivers/csl/iss/iss_common/iss_common.h"
-#include "wtsd_ducatimmsw/drivers/csl/iss/simcop/common/simcop_irq.h"
-#include "wtsd_ducatimmsw/alg/lbce/msp_lbce.h"
+#include "WTSD_DucatiMMSW/framework/msp/msp.h"
+#include "WTSD_DucatiMMSW/framework/resource_manager/rm.h"
+#include  "WTSD_DucatiMMSW/drivers/csl/iss/iss_common/iss_common.h"
+#include "WTSD_DucatiMMSW/drivers/csl/iss/simcop/common/simcop_irq.h"
+#include "WTSD_DucatiMMSW/alg/lbce/msp_lbce.h"
 #include <xdc/runtime/Timestamp.h>
 
 
