Index: tisocketnode-mpeg4aacdec-4.0+cc+DSP-MM-TESLA_RLS_4.1/audio/node/mpeg4aac/dec/makefile
===================================================================
--- tisocketnode-mpeg4aacdec-4.0+cc+DSP-MM-TESLA_RLS_4.1.orig/audio/node/mpeg4aac/dec/makefile	2009-09-03 22:17:36.000000000 +0200
+++ tisocketnode-mpeg4aacdec-4.0+cc+DSP-MM-TESLA_RLS_4.1/audio/node/mpeg4aac/dec/makefile	2009-09-03 22:18:36.000000000 +0200
@@ -10,7 +10,7 @@
 
 
 # Define version of trees to import from.  
-include $(DSPMAKEROOT)\product.mak
+include $(DSPMAKEROOT)/product.mak
 
 # This ensures that sm-make only calls us once.  We handle DEBUG flags
 # internally on our own
@@ -115,7 +115,7 @@
 # build verbose-ly, using the appropriate config.bld, and setting the package
 # path to include the imports directory (where packages are exploded into)
 DD_XDCOPTIONS = XDCOPTIONS=v XDCBUILDCFG=$(DSPMAKEROOT)/config.bld \
-    XDCPATH=$(SABIOS_DIR)\;$(DEPOT)/framework_components_$(FC_VER)/fctools/packages\;$(BRIDGETOOL_DIR)/packages\;$(FC_DIR)\;$(CURDIR)\;$(DEPOT)/osal_$(OSAL_VER)/packages\;$(DEPOT)/ipc_$(IPC_VER)/packages\;$(CURDIR)/inc/;$(DSPMAKEROOT)/../system/utils/inc\;$(DSPMAKEROOT)/../system/inst2/inst2_lib/inc\;$(DBS_BRIDGE_DIR_C64)/ti/dspbridge/dsp/bridge_product/exports/include\;$(DBS_BRIDGE_DIR_C64);$(DSPMAKEROOT)/../system/usn/inc;$(DSPMAKEROOT)/../audio/alg/aacdec/omap4_he_mpeg4/inc \
+    XDCPATH=$(SABIOS_DIR)\;$(DEPOT)/framework_components_$(FC_VER)/fctools/packages\;$(BRIDGETOOL_DIR)/packages\;$(FC_DIR)\;$(CURDIR)\;$(DEPOT)/osal_$(OSAL_VER)/packages\;$(DEPOT)/ipc_$(IPC_VER)/packages\;$(CURDIR)/inc\;$(DSPMAKEROOT)/../system/utils/inc\;$(DSPMAKEROOT)/../system/inst2/inst2_lib/inc\;$(DBS_BRIDGE_DIR_C64)/ti/dspbridge/dsp/bridge_product/exports/include\;$(DBS_BRIDGE_DIR_C64)\;$(DSPMAKEROOT)/../system/usn/inc\;$(DSPMAKEROOT)/../audio/alg/aacdec/omap4_he_mpeg4/inc \
     XDCTARGETS="$(XDCTARGETS)"
 
 SAMPLE_PACKAGES = src 
Index: tisocketnode-mpeg4aacdec-4.0+cc+DSP-MM-TESLA_RLS_4.1/audio/node/mpeg4aac/dec/src/mpeg4aacdec_sn.cfg
===================================================================
--- tisocketnode-mpeg4aacdec-4.0+cc+DSP-MM-TESLA_RLS_4.1.orig/audio/node/mpeg4aac/dec/src/mpeg4aacdec_sn.cfg	2009-09-03 22:20:10.000000000 +0200
+++ tisocketnode-mpeg4aacdec-4.0+cc+DSP-MM-TESLA_RLS_4.1/audio/node/mpeg4aac/dec/src/mpeg4aacdec_sn.cfg	2009-09-03 22:20:23.000000000 +0200
@@ -19,7 +19,7 @@
  *
  */
 
-xdc.loadCapsule("../../../../../vobs_SDS_Source_Bridge_dsp/ti/dspbridge/dsp/bridge_config/platforms/c64T_dyn.cfg");
+xdc.loadCapsule("ti/dspbridge/dsp/bridge_config/platforms/c64T_dyn.cfg");
 
 var MPEG4AACDEC_SN 		= new Node.Params();
 MPEG4AACDEC_SN.uuid 		= "5C89A1F1_3D83_11D6_B0D7_00C04F1FC036";
