Index: tisocketnode-usn-4.0+cc+DSP-MM-TESLA_RLS_4.1/system/usn/makefile
===================================================================
--- tisocketnode-usn-4.0+cc+DSP-MM-TESLA_RLS_4.1.orig/system/usn/makefile	2009-09-03 23:31:01.000000000 +0200
+++ tisocketnode-usn-4.0+cc+DSP-MM-TESLA_RLS_4.1/system/usn/makefile	2009-09-03 23:31:37.000000000 +0200
@@ -10,7 +10,7 @@
 
 
 # Define version of trees to import from.  
-include $(DSPMAKEROOT)\product.mak
+include $(DSPMAKEROOT)/product.mak
 
 # This ensures that sm-make only calls us once.  We handle DEBUG flags
 # internally on our own
@@ -115,7 +115,7 @@
 # build verbose-ly, using the appropriate config.bld, and setting the package
 # path to include the imports directory (where packages are exploded into)
 DD_XDCOPTIONS = XDCOPTIONS=v XDCBUILDCFG=$(DSPMAKEROOT)/config.bld \
-    XDCPATH=$(SABIOS_DIR)\;$(DEPOT)/framework_components_$(FC_VER)/fctools/packages\;$(BRIDGETOOL_DIR)/packages\;$(FC_DIR)\;$(CURDIR)\;$(DEPOT)/osal_$(OSAL_VER)/packages\;$(DEPOT)/ipc_$(IPC_VER)/packages\;$(CURDIR)/inc\;$(DSPMAKEROOT)/../system/utils/inc\;$(DSPMAKEROOT)/../system/inst2/inst2_lib/inc\;$(DBS_BRIDGE_DIR_C64)/ti/dspbridge/dsp/bridge_product/exports/include\;$(DBS_BRIDGE_DIR_C64);$(DSPMAKEROOT)/../system/dasf/MixerDeviceNode/inc; \
+    XDCPATH=$(SABIOS_DIR)\;$(DEPOT)/framework_components_$(FC_VER)/fctools/packages\;$(BRIDGETOOL_DIR)/packages\;$(FC_DIR)\;$(CURDIR)\;$(DEPOT)/osal_$(OSAL_VER)/packages\;$(DEPOT)/ipc_$(IPC_VER)/packages\;$(CURDIR)/inc\;$(DSPMAKEROOT)/../system/utils/inc\;$(DSPMAKEROOT)/../system/inst2/inst2_lib/inc\;$(DBS_BRIDGE_DIR_C64)/ti/dspbridge/dsp/bridge_product/exports/include\;$(DBS_BRIDGE_DIR_C64)\;$(DSPMAKEROOT)/../system/dasf/MixerDeviceNode/inc \
     XDCTARGETS="$(XDCTARGETS)"
 
 SAMPLE_PACKAGES = src 
Index: tisocketnode-usn-4.0+cc+DSP-MM-TESLA_RLS_4.1/system/usn/src/usn_dll.cfg
===================================================================
--- tisocketnode-usn-4.0+cc+DSP-MM-TESLA_RLS_4.1.orig/system/usn/src/usn_dll.cfg	2009-09-03 23:31:54.000000000 +0200
+++ tisocketnode-usn-4.0+cc+DSP-MM-TESLA_RLS_4.1/system/usn/src/usn_dll.cfg	2009-09-03 23:32:09.000000000 +0200
@@ -18,6 +18,6 @@
  *  ======== usn.cfg ========
  *
  */
-xdc.loadCapsule("../../../../../vobs_SDS_Source_Bridge_dsp/ti/dspbridge/dsp/bridge_config/platforms/c64T_dyn.cfg");
+xdc.loadCapsule("ti/dspbridge/dsp/bridge_config/platforms/c64T_dyn.cfg");
 
 
