Index: tidspbridge-dllcreate-24.0.6/DLLcreate/src/header.h
===================================================================
--- src.orig/DLLcreate/src/header.h 2008-11-14 12:51:01.000000000 -0600
+++ src/DLLcreate/src/header.h      2008-11-14 12:51:11.000000000 -0600
@@ -6,6 +6,9 @@
 /*                                                                     */
 /***********************************************************************/

+
+#include <string.h>
+
 /*-----------------------------------------------------------------*/
 /* INCLUDE HEADER FILES FOR RUNTIME SUPPORT.                       */
 /*-----------------------------------------------------------------*/
@@ -84,8 +87,10 @@
 #elif defined(__GNUC__)
 
   typedef char* string;
+/*
   #include <iostream>
   #include <iomanip>
+*/
   #include <list>
   #include <vector>
   #include <deque>
