--- /scripts/install_tesla_bridge.orig	2009-09-02 11:42:48.000000000 -0500
+++ /scripts/install_tesla_bridge	2009-09-02 11:43:49.000000000 -0500
@@ -5,16 +5,16 @@
 echo " "
 echo " "
 #install omap_notify driver
-echo "installing /binaries/omap_notify.ko"
-insmod /binaries/omap_notify.ko
+echo "installing omap_notify.ko"
+insmod omap_notify.ko
 
 #install notify syslink proc driver
 echo "installing notify_tesladriver.ko"
-insmod /binaries/notify_tesladriver.ko
+insmod notify_tesladriver.ko
 
 #install notify syslink proc4430 driver
 echo "installing dspbridge.ko"
-insmod /binaries/dspbridge.ko
+insmod dspbridge.ko
 
 #install notify tesla driver
 echo "installing bridgedriver.ko"
