--- /scripts/install_ducati_syslink.orig	2009-09-02 11:38:18.000000000 -0500
+++ /scripts/install_ducati_syslink	2009-09-02 11:39:13.000000000 -0500
@@ -5,12 +5,12 @@
 echo " "
 echo " "
 #install omap_notify driver
-echo "installing /binaries/omap_notify.ko"
-insmod /binaries/omap_notify.ko
+echo "installing omap_notify.ko"
+insmod omap_notify.ko
 
 #install notify syslink proc driver
 echo "installing syslink_proc.ko"
-insmod /binaries/syslink_proc.ko
+insmod syslink_proc.ko
 
 #install notify syslink proc4430 driver
 echo "installing syslink_proc4430.ko"
