--- tidspbridge-syslinklib-24.0.9/scripts/install_tesla_bridge.orig	2009-08-14 16:26:35.000000000 -0500
+++ tidspbridge-syslinklib-24.0.9/scripts/install_tesla_bridge	2009-08-14 16:29:53.000000000 -0500
@@ -5,20 +5,20 @@
 echo " "
 echo " "
 #install omap_notify driver
-echo "installing /binaries/omap_notify.ko"
-insmod /binaries/omap_notify.ko
+echo "installing omap_notify.ko"
+insmod omap_notify.ko
 
 #install notify dispatcher driver
-echo "installing notify_disp.ko"
-insmod /binaries/notify_disp.ko
+#echo "installing notify_disp.ko"
+#insmod notify_disp.ko
 
 #install notify syslink proc driver
 echo "installing notify_tesladriver.ko"
-insmod /binaries/notify_tesladriver.ko
+insmod notify_tesladriver.ko
 
 #install notify syslink proc4430 driver
 echo "installing dspbridge.ko"
-insmod /binaries/dspbridge.ko
+insmod dspbridge.ko
 
 #install notify tesla driver
 echo "installing bridgedriver.ko"
