--- /scripts/install_syslink.orig	2009-09-02 11:41:04.000000000 -0500
+++ /scripts/install_syslink	2009-09-02 11:41:33.000000000 -0500
@@ -5,12 +5,12 @@
 echo " "
 echo " "
 #install omap_notify driver
-echo "installing /binaries/omap_notify.ko"
-insmod /binaries/omap_notify.ko
+echo "installing omap_notify.ko"
+insmod omap_notify.ko
 
 #install notify syslink proc driver
 echo "installing syslink_proc.ko"
-insmod /binaries/syslink_proc.ko
+insmod syslink_proc.ko
 
 #install syslink ipc driver
 echo "installing syslink_ipc.ko"
@@ -27,11 +27,11 @@
 
 #install notify syslink proc driver
 echo "installing notify_tesladriver.ko"
-insmod /binaries/notify_tesladriver.ko
+insmod notify_tesladriver.ko
 
 #install notify syslink proc4430 driver
 echo "installing dspbridge.ko"
-insmod /binaries/dspbridge.ko
+insmod dspbridge.ko
 
 
 #install notify tesla driver
